bottlecap/src/lifecycle/invocation/triggers/mod.rs

```diff
@@ -31,6 +50,45 @@ pub trait ServiceNameResolver {
     fn get_generic_identifier(&self) -> &'static str;
 }
 
+#[must_use]
+pub fn parameterize_api_resource(resource: String) -> String {
+    // curly braces are used for APIGW parameters feature
+    if resource.contains('{') && resource.contains('}') {
+        return resource;
+    }
+
+    let parts: Vec<&str> = resource.split('/').collect();
+    let mut result = Vec::new();
+
+    // First element is empty string due to leading slash
+    result.push(String::new());
+
+    // Process each path segment
+    for (i, part) in parts.iter().enumerate().skip(1) {
+        if part.is_empty() {
+            continue;
+        }
+
+        // Check if this part looks like an identifier
+        // Number, ULID, GUID, or UUID
+        if part.chars().all(|c| c.is_ascii_digit()) || ULID_UUID_GUID.is_match(part) {
+            // Determine the parameter name based on the previous segment
+            let param_name = if i > 1 && !parts[i - 1].is_empty() {
+                let singular = parts[i - 1].trim_end_matches('s');
+                format!("{singular}_id")
```

Comment:
Do we want to name id `_id`? 

Are we sure we are going to catch everything? If it's not a UUID, GUID, ULID, it won't catch any other stuff, this won't effectively reduce cardinality, right? I guess it will reduce it substantially tho

Reply:
Let me write a test/catch `/id/12434`, but yeah – this won't fix everything. At some point we'll want to expose a configurable env var for this